x-pack/plugins/lens/public/editor_frame_plugin/editor_frame/editor_frame.test.tsx

```diff
@@ -439,4 +421,284 @@ describe('editor_frame', () => {
       );
     });
   });
+
+  describe('suggestions', () => {
+    it('should fetch suggestions of currently active datasource', async () => {
+      mount(
+        <EditorFrame
+          visualizationMap={{
+            testVis: mockVisualization,
+          }}
+          datasourceMap={{
+            testDatasource: mockDatasource,
+            testDatasource2: mockDatasource2,
+          }}
+          initialDatasourceId="testDatasource"
+          initialVisualizationId="testVis"
+        />
+      );
+
+      await waitForPromises();
+
+      expect(mockDatasource.getDatasourceSuggestionsFromCurrentState).toHaveBeenCalled();
+      expect(mockDatasource2.getDatasourceSuggestionsFromCurrentState).not.toHaveBeenCalled();
+    });
+
+    it('should fetch suggestions of all visualizations', async () => {
+      mount(
+        <EditorFrame
+          visualizationMap={{
+            testVis: mockVisualization,
+            testVis2: mockVisualization2,
+          }}
+          datasourceMap={{
+            testDatasource: mockDatasource,
+            testDatasource2: mockDatasource2,
+          }}
+          initialDatasourceId="testDatasource"
+          initialVisualizationId="testVis"
+        />
+      );
+
+      await waitForPromises();
+
+      expect(mockVisualization.getSuggestions).toHaveBeenCalled();
+      expect(mockVisualization2.getSuggestions).toHaveBeenCalled();
+    });
+
+    it('should display suggestions in descending order', async () => {
+      const instance = mount(
+        <EditorFrame
+          visualizationMap={{
+            testVis: {
+              ...mockVisualization,
+              getSuggestions: () => [
+                {
+                  datasourceSuggestionId: 0,
+                  score: 0.5,
+                  state: {},
+                  title: 'Suggestion2',
+                },
+                {
+                  datasourceSuggestionId: 0,
```

Comment:
You're reusing suggestion IDs here, I would expect a uniqueness error.

Reply:
This is the implementation of getSuggestions of the visualization, not the datasource. These ids are referencing the datatable used for this visualization suggestion. A single visualization can provide multiple suggestions based on a single datatable, this is not an error. Because this concept is pretty misleading, I changed the naming (see your other comment down below)